client/components/convert-to-blocks/index.jsx

```diff
@@ -1,62 +1,67 @@
 /**
  * External dependencies
  */
-import PropTypes from 'prop-types';
-import React, { Component } from 'react';
-import { localize } from 'i18n-calypso';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import { Dialog } from '@automattic/components';
+import isConvertToBlocksDialogDismissed from 'state/selectors/is-convert-to-blocks-dialog-dismissed';
+import { dismissConvertToBlocksDialog } from 'state/ui/convert-to-blocks-dialog/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
 
 /**
  * Style dependencies
  */
+import '@wordpress/components/build-style/style.css';
 import './style.scss';
 
-class ConvertToBlocksDialog extends Component {
-	static propTypes = {
-		showDialog: PropTypes.bool,
-		handleConversionResponse: PropTypes.func,
-	};
-
-	close = ( action ) => {
-		this.props.handleResponse( action === 'convert' );
-	};
+export default function ConvertToBlocksDialog( { handleResponse, postId, showDialog } ) {
+	const { isDismissed, siteId } = useSelector( ( state ) => {
+		const selectedSiteId = getSelectedSiteId( state );
+		return {
+			isDismissed: isConvertToBlocksDialogDismissed( state, selectedSiteId, postId ),
+			siteId: selectedSiteId,
+		};
+	} );
+	const dispatch = useDispatch();
+	const translate = useTranslate();
 
-	render() {
-		const { translate, showDialog } = this.props;
-
-		const buttons = [
-			{
-				action: 'convert',
-				isPrimary: true,
-				label: translate( 'Convert to Blocks' ),
-			},
-			{
-				action: 'cancel',
-				label: translate( 'Cancel' ),
-			},
-		];
+	if ( ! showDialog || isDismissed ) {
+		return null;
+	}
 
-		return (
-			<Dialog
-				additionalClassNames="editor__gutenberg-convert-blocks-dialog"
-				buttons={ buttons }
-				isVisible={ showDialog }
-				onClose={ this.close }
-			>
-				<h1>{ translate( 'Ready to try blocks?' ) }</h1>
+	const onClose = ( action ) => () => {
+		handleResponse( action === 'convert' );
+		dispatch( dismissConvertToBlocksDialog( siteId, postId ) );
+	};
 
-				<p>
-					{ translate(
-						'This post contains content you created using the older editor. For the best editing experience, we recommend converting this content to blocks.'
-					) }
-				</p>
-			</Dialog>
-		);
-	}
+	return (
+		<Modal
+			onRequestClose={ onClose( 'cancel' ) }
+			shouldCloseOnClickOutside={ false }
+			title={ translate( 'Convert to blocks?' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'Convert to blocks' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/convert-to-blocks-dialog)